File: app/src/main/java/tech/inscripts/ins_armman/mMitra/completedFormDetails/DetailsAdapter.java
package tech.inscripts.ins_armman.mMitra.completedFormDetails;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.inscripts.ins_armman.npdsf.R;
import com.inscripts.ins_armman.npdsf.data.model.CompleteFormQnA;
import com.inscripts.ins_armman.npdsf.utility.utility;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class DetailsAdapter extends RecyclerView.Adapter<DetailsAdapter.ViewHolder> {
    Context mContext;
    String question, answer, language;
    int formId = 0;
    private ArrayList<CompleteFormQnA> mDetails;

    public DetailsAdapter(ArrayList<CompleteFormQnA> mDetails, Context mContext, int form_id) {
        this.mDetails = mDetails;
        this.mContext = mContext;
        this.formId = form_id;
    }


    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.activity_list_item, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        CompleteFormQnA c = mDetails.get(position);
        answer = c.getAnswer();
        question=c.getQuestion();
        if ((answer.contains("{") && question.contains("{")) ) {
            try {
                JSONObject obj = new JSONObject(c.getQuestion());
                JSONObject obj1 = new JSONObject(c.getAnswer());
                language = utility.getLanguagePreferance(mContext);
        if (language.isEmpty()) {
                    utility.setApplicationLocale(mContext, "en");
                } else {
                    utility.setApplicationLocale(mContext, language);
                }
                question = obj.getString(this.language);
            answer = obj1.getString(this.language);
                holder.txtQuestion.setText(question);
                holder.txtAnswer.setText(answer);
                System.out.println("settext question :" + question);
                System.out.println("settext answer :" + answer);

            } catch (JSONException e) {
                e.printStackTrace();
            }

        }
        else if(question.contains("{") && !answer.contains("{") ){
            try {
                JSONObject objQue = new JSONObject(c.getQuestion());
                language = utility.getLanguagePreferance(mContext);
                if (language.isEmpty()) {
                    utility.setApplicationLocale(mContext, "en");
                } else {
                    utility.setApplicationLocale(mContext, language);
                }
                question = objQue.getString(this.language);
                holder.txtQuestion.setText(question);
                holder.txtAnswer.setText(answer);

            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
        else if(!question.contains("{") && answer.contains("{") ) {
            try {
                JSONObject objAns = new JSONObject(c.getAnswer());
                language = utility.getLanguagePreferance(mContext);
                if (language.isEmpty()) {
                    utility.setApplicationLocale(mContext, "en");
                } else {
                    utility.setApplicationLocale(mContext, language);
                }
                answer = objAns.getString(this.language);
                holder.txtQuestion.setText(question);
                holder.txtAnswer.setText(answer);
            } catch (JSONException e) {
                e.printStackTrace();
            }

        }
        else {
            holder.txtQuestion.setText(question);
            holder.txtAnswer.setText(answer);
        }
    }

    @Override
    public int getItemCount() {
        return mDetails.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView txtQuestion, txtAnswer;

        public ViewHolder(View itemView) {
            super(itemView);
            txtQuestion = itemView.findViewById(R.id.tvQuestion);
            txtAnswer = itemView.findViewById(R.id.tvAnswer);
        }
    }
}

File: app/src/main/java/tech/inscripts/ins_armman/mMitra/completeformsdetails/DetailsAdapter.kt
package tech.inscripts.ins_armman.mMitra.completeformsdetails

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import org.json.JSONException
import org.json.JSONObject
import tech.inscripts.ins_armman.mMitra.R
import tech.inscripts.ins_armman.mMitra.data.model.CompleteFormQnA
import tech.inscripts.ins_armman.mMitra.utility.Utility
import java.util.ArrayList

class DetailsAdapter() : RecyclerView.Adapter<DetailsAdapter.ViewHolder>() {

    lateinit var mContext : Context
    var question : String?=null
    var answer : String?=null
    var language : String?=null
    var formId: Int = 0
    var utility= Utility()
    private var mDetails: ArrayList<CompleteFormQnA>?=null

    constructor(mContext: Context, formId: Int, mDetails: ArrayList<CompleteFormQnA>?) : this() {
        this.mContext = mContext
        this.formId = formId
        this.mDetails = mDetails
    }


    override fun onCreateViewHolder(viewGroup: ViewGroup, p1: Int): DetailsAdapter.ViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.activity_list_item, viewGroup, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return mDetails?.size!!
    }

    override fun onBindViewHolder(holder: DetailsAdapter.ViewHolder, position: Int) {
    var c : CompleteFormQnA = mDetails?.get(position)!!
        answer = c.answer
        question=c.question
        if (answer!!.contains("{") && question!!.contains("{")) {
            try {
                val obj = JSONObject(c.question)
                val obj1 = JSONObject(c.answer)
                language = utility.getLanguagePreferance(mContext)
                if (language!!.isEmpty()) {
                    utility.setApplicationLocale(mContext, "en")
                } else {
                    utility.setApplicationLocale(mContext, language!!)
                }
                question = obj.getString(this.language)
                answer = obj1.getString(this.language)
                holder.txtQuestion.text = question
                holder.txtAnswer.text = answer
                println("settext question :$question")
                println("settext answer :$answer")

            } catch (e: JSONException) {
                e.printStackTrace()
            }

        } else if (question!!.contains("{") && !answer!!.contains("{")) {
            try {
                val objQue = JSONObject(c.question)
                language = utility.getLanguagePreferance(mContext)
                if (language!!.isEmpty()) {
                    utility.setApplicationLocale(mContext, "en")
                } else {
                    utility.setApplicationLocale(mContext, language!!)
                }
                question = objQue.getString(this.language)
                holder.txtQuestion.text = question
                holder.txtAnswer.text = answer

            } catch (e: JSONException) {
                e.printStackTrace()
            }

        } else if (!question!!.contains("{") && answer!!.contains("{")) {
            try {
                val objAns = JSONObject(c.answer)
                language = utility.getLanguagePreferance(mContext)
                if (language!!.isEmpty()) {
                    utility.setApplicationLocale(mContext, "en")
                } else {
                    utility.setApplicationLocale(mContext, language!!)
                }
                answer = objAns.getString(this.language)
                holder.txtQuestion.text = question
                holder.txtAnswer.text = answer
            } catch (e: JSONException) {
                e.printStackTrace()
            }

        } else {
            holder.txtQuestion.text = question
            holder.txtAnswer.text = answer
        }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var txtQuestion: TextView
        internal var txtAnswer: TextView

        init {
            txtQuestion = itemView.findViewById(R.id.tvQuestion)
            txtAnswer = itemView.findViewById(R.id.tvAnswer)
        }
    }
}